runtime/j9vm/j9scar.tdf

```diff
@@ -351,17 +351,17 @@ TraceEvent=Trc_SC_LoadLibrary_OpenShared_Decorate NoEnv Overhead=1 Level=3 Templ
 
 TraceExit=Trc_SC_LoadLibrary_Exit NoEnv Overhead=1 Level=2 Template="JVM_LoadLibrary -- return 0x%zx"
 
-TraceEntry=Trc_SC_VirtualThreadMountBegin_Entry Overhead=1 Level=3 Template="thread = 0x%p, firstMount = %d"
-TraceExit=Trc_SC_VirtualThreadMountBegin_Exit Overhead=1 Level=3 Template="thread = 0x%p, firstMount = %d"
+TraceEntry=Trc_SC_VirtualThreadMountBegin_Entry Obsolete Overhead=1 Level=3 Template="thread = 0x%p, firstMount = %d"
+TraceExit=Trc_SC_VirtualThreadMountBegin_Exit Obsolete Overhead=1 Level=3 Template="thread = 0x%p, firstMount = %d"
 
-TraceEntry=Trc_SC_VirtualThreadMountEnd_Entry Overhead=1 Level=3 Template="thread = 0x%p, firstMount = %d"
-TraceExit=Trc_SC_VirtualThreadMountEnd_Exit Overhead=1 Level=3 Template="thread = 0x%p, firstMount = %d"
+TraceEntry=Trc_SC_VirtualThreadMountEnd_Entry Obsolete Overhead=1 Level=3 Template="thread = 0x%p, firstMount = %d"
+TraceExit=Trc_SC_VirtualThreadMountEnd_Exit Obsolete Overhead=1 Level=3 Template="thread = 0x%p, firstMount = %d"
 
-TraceEntry=Trc_SC_VirtualThreadUnmountBegin_Entry Overhead=1 Level=3 Template="thread = 0x%p, lastUnmount = %d"
-TraceExit=Trc_SC_VirtualThreadUnmountBegin_Exit Overhead=1 Level=3 Template="thread = 0x%p, lastUnmount = %d"
+TraceEntry=Trc_SC_VirtualThreadUnmountBegin_Entry Obsolete Overhead=1 Level=3 Template="thread = 0x%p, lastUnmount = %d"
+TraceExit=Trc_SC_VirtualThreadUnmountBegin_Exit Obsolete Overhead=1 Level=3 Template="thread = 0x%p, lastUnmount = %d"
 
-TraceEntry=Trc_SC_VirtualThreadUnmountEnd_Entry Overhead=1 Level=3 Template="thread = 0x%p, lastUnmount = %d"
-TraceExit=Trc_SC_VirtualThreadUnmountEnd_Exit Overhead=1 Level=3 Template="thread = 0x%p, lastUnmount = %d"
+TraceEntry=Trc_SC_VirtualThreadUnmountEnd_Entry Obsolete Overhead=1 Level=3 Template="thread = 0x%p, lastUnmount = %d"
+TraceExit=Trc_SC_VirtualThreadUnmountEnd_Exit Obsolete Overhead=1 Level=3 Template="thread = 0x%p, lastUnmount = %d"
 
```

Comment:
I think we need both the old and new tracepoints (the old ones should be used for Java 20).

Reply:
- The new tracepoints are not needed.
- The old tracepoints can also be used in JDK21. In JDK21, we implicitly know that `JVM_VirtualThreadStart (firstMount=true)`, `JVM_VirtualThreadEnd (lastUnmount=true)`, `JVM_VirtualThreadMount(firstMount=false)` and `JVM_VirtualThreadUnmount (lastUnmount=false)`.
- In JDK20, the new tracepoints remove information about `firstMount` and `lastUnmount`. This information should be retained since it is valuable for debugging. We can pass `firstMount` and `lastUnmount` values to the new `virtualThread*` helper functions and keep using the old tracepoints.

Reply:
There are two commits in the PR. With the above feedback, we will revert some elements to the first commit. If we decide to go with this feedback, my suggestion will be to squash and summarize all changes in a single commit to avoid confusion.

Reply:
After taking another look, the tracepoints should be added in the `JVM_*` functions, not in the `virtualThread*` helper functions, since the `_Entry` and `_Exit` tracepoints represent the entry and exit points of the `JVM_*` functions. This also implies that we should add new tracepoints: they should be named as per the new `JVM_*` functions (re: https://github.com/eclipse-openj9/openj9/pull/17125#discussion_r1220147949) and they should include all input parameters to the `JVM_*` functions.

Reply:
@babsingh reverted to old tracepoints and added tracepoints for new functions.